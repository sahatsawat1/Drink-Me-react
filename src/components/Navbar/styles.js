import styled from 'styled-components'

export const Container = styled.div`
  width: 100%;
  /* position: ${props => (props.isScroll ? 'fixed' : 'static')}; */
  position: fixed;
  background-color: ${props => (props.isScroll ? 'rgba(28, 29, 34, 0.6)' : 'rgba(28, 29, 34, .9)')};
  color: ${props => (props.isScroll ? '#DFCFAF' : '#DFCFAF')};
  transition: background-color .5s;
`
export const Wrapper = styled.div`
  display: flex;
  padding: 2rem;
  flex: 1;
  justify-content: center;
  align-items: center;
  a{
    width: 10%;
  }
  img{
    width: 100%;
  }
`

export const ActionContainer = styled.div`
  display: flex;
  flex: 1;
  justify-content: flex-start;
  a{
    border-right: 1px solid #DFCFAF;
    padding: 0 3rem;
  }
`
export const Action = styled.li`
  list-style-type: none;
`
export const LoginContainer =styled.div`
  font-size: 24px;
  font-weight: 900;
`