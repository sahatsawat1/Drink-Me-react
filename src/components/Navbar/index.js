import React from 'react'
import useScroll from '../../utils/useScroll'
import { Link } from 'react-router-dom'
import { Container, Wrapper, Logo, ActionContainer, Action, LoginContainer } from './styles'

function Navbar () {
  const isScroll = useScroll({ scrollRange: 20 })

  return (
    <Container isScroll={isScroll}>
      <Wrapper>
        <Logo>LOGO</Logo>
        <ActionContainer>
          <Action>
            <Link to='/'>HOME</Link>
          </Action>
          <Action>
            <Link to='/about'>TABLE</Link>
          </Action>
          <Action>
            <Link to='/contact'>BILL</Link>
          </Action>
        </ActionContainer>
        <LoginContainer>
          <Action>
            <Link to='/contact'>LOGIN</Link>
          </Action>
        </LoginContainer>
      </Wrapper>
    </Container>
  )
}
export default Navbar
