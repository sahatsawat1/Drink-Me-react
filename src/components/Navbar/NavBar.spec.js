/* global describe test expect */
import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import { render } from '@testing-library/react'
import Navbar from '.'

describe('Navbar Component', () => {
  test('should contain logo text', () => {
    const { getByText } = render(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>
    )
    expect(getByText(/logo/gi)).toBeInTheDocument()
  })
  test('should have link navigate to home ,about,contact,and story', () => {
    const { getByText } = render(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>
    )
    expect(getByText(/home/gi)).toBeInTheDocument()
    expect(getByText(/home/gi).closest('a')).toHaveAttribute('href', '/')
    expect(getByText(/about/gi)).toBeInTheDocument()
    expect(getByText(/about/gi).closest('a')).toHaveAttribute('href', '/about')
    expect(getByText(/contact/gi)).toBeInTheDocument()
    expect(getByText(/contact/gi).closest('a')).toHaveAttribute('href', '/contact')
    expect(getByText(/story/gi)).toBeInTheDocument()
    expect(getByText(/story/gi).closest('a')).toHaveAttribute('href', '/story')
  })
  test('should render layout correctly', () => {
    const { container } = render(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>
    )
    expect(container).toMatchSnapshot()
  })
})
