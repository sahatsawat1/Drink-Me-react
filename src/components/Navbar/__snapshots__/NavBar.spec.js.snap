// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Component should render layout correctly 1`] = `
<div>
  <div
    class="sc-bdnylx hsSgNd"
  >
    <div
      class="sc-gtssRu gIgNkG"
    >
      <div
        class="sc-dlnjPT"
      >
        Logo
      </div>
      <div
        class="sc-hKFyIo ljvqsi"
      >
        <li
          class="sc-eCApGN evAwKw"
        >
          <a
            href="/"
          >
            Home
          </a>
        </li>
        <li
          class="sc-eCApGN evAwKw"
        >
          <a
            href="/about"
          >
            About
          </a>
        </li>
        <li
          class="sc-eCApGN evAwKw"
        >
          <a
            href="/contact"
          >
            Contact
          </a>
        </li>
        <li
          class="sc-eCApGN evAwKw"
        >
          <a
            href="/story"
          >
            Story
          </a>
        </li>
      </div>
    </div>
  </div>
</div>
`;
